pkg/crc/machine/vsockprivateaddress_windows.go

```diff
@@ -0,0 +1,65 @@
+package machine
+
+import (
+	"errors"
+	"net"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+)
+
+const (
+	fallbackPrivateAddress = "127.0.0.1"
+	adapterName            = "vEthernet (WSL)"
+	ipv4Prefix             = "172."
+)
+
+// On Windows it would be useful if WSL2 can access VSock resources as well
+// This address should work on both the windows side and the WSL2 side
+
+var (
+	vsockCheckedWsl          = false
+	vsockPrivateAddressToUse = "127.0.0.1"
+)
+
+func VsockPrivateAddress() (addrStr string) {
+	if vsockCheckedWsl {
```

Comment:
This really looks cryptic, in first glance it looks like in case vsock checked with wsl and it true then return `127.0.0.1` . I think this function need to re looked around what it try to do.

```golang
func VsockPrivateAddress() string {
        addrResult, err := localhostOrWslhost()
	if err != nil {
		logging.Warnf("Encountered error trying to resolve WSL host IP: %s", err)
	}
        return addrResult
}
```

Reply:
`fallbackPrivateAddress` need to be returned here, when `wslhostAddress()` (current: `localhostOrWslhost()`) would return an `err`.

This will also make the toggle much easier in a single place. No need to jump into `wslhostAddress` when the config is not set.

Reply:
IIUC these two functions are added to find out the ip address of the WSL specific virtual switch named `vEthernet (WSL)` if that is correct, then we can simplify it a bit by using powershell commands to get the IP address, e.g. `(Get-NetIPAddress -AddressFamily IPv4 -InterfaceAlias "vEthernet (WSL)").IPAddress`